lib/update.rb

```diff
@@ -157,33 +159,16 @@ def self.download_update
         else
           _respond; _respond "Downloading Lich5 version #{@update_to}"; _respond
           filename = "lich5-#{@update_to}"
-          File.open(File.join(TEMP_DIR, "#{filename}.zip"), "wb") do |file|
+          File.open(File.join(TEMP_DIR, "#{filename}.tar.gz"), "wb") do |file|
             file.write open("#{@zipfile}").read
           end
 
-          if defined?(Win32)
-            Dir.chdir(TEMP_DIR)
-            system("tar -xf #{File.join(TEMP_DIR, filename)}.zip")
-            sleep 2
-            system("move elanthia-online-lich* #{filename}")
-            sleep 2
-          else
-            finish = spawn "unzip #{File.join(TEMP_DIR, filename)}.zip -d #{File.join(TEMP_DIR, filename)}"
-          end
-          unless defined?(Win32)
-            50.times { break unless (Process.waitpid(finish, Process::WNOHANG)).nil?; sleep 0.1 }
-
-            new_target = Dir.children(File.join(TEMP_DIR, filename))
-            FileUtils.cp_r(File.join(TEMP_DIR, filename, new_target[0]), TEMP_DIR)
-            FileUtils.remove_dir(File.join(TEMP_DIR, filename))
-            FileUtils.mv(File.join(TEMP_DIR, new_target[0]), File.join(TEMP_DIR, filename))
-          end
-          ## These five lines may be deleted after zipfiles are created with .gitattributes
-          cleanup_dir = Dir.children(File.join(TEMP_DIR, filename))
-          FileUtils.remove_dir(File.join(TEMP_DIR, filename, "lich")) if cleanup_dir.include?("lich")
-          FileUtils.rm(File.join(TEMP_DIR, filename, "netlify.toml")) if cleanup_dir.include?("netlify.toml")
-          FileUtils.rm(File.join(TEMP_DIR, filename, "README.adoc")) if cleanup_dir.include?("README.adoc")
-          FileUtils.rm(File.join(TEMP_DIR, filename, "data", "update-lich5.json")) if Dir.children(File.join(TEMP_DIR, filename, "data")).include?("update-lich5.json")
+          Dir.mkdir("#{TEMP_DIR}/#{filename}")
```

Comment:
this should use `File.join`

Reply:
So. . . 

`Dir.mkdir(File.join(TEMP_DIR, filename))`

I'll test it out - no issue with the change.